exercise_004_extension_methods/README.md

```diff
@@ -4,10 +4,59 @@
 
 Extension methods can be used to add methods to types after they are defined.
 
-## Steps
+Lets say we need to add a method `square` to type `Int`. We used to achieve it 
+with the help of `implicit class`es in _Scala 2_
+
+```scala
+implicit class RichInt(val i: Int) extends AnyVal {
+  def square = i * i
+}
+```
+
+Now you can use it as follows
+
+```scala
+4.square
+// res1: Int = 16
+```
+
+With dotty `extension methods`, we can rewrite the above as
+
+```scala
+def (i: Int).square: Int = i * i
+
+3.square
+// val res0: Int = 9
+```
+
+More than one extension methods can be wrapped inside an `Extension Instance`.
+```scala
+extension {
+  def (x: Int).square : Int = x * x
+  def (x: Int).isEven: Boolean = x % 2 == 0
+}
+```
+
+`Collective Extensions` can be used if all your extension methods share the 
+same left-hand parameter type.
```

Comment:
I would add that this restriction is specific to Scala 3.0 and it will be lifted in a later Scala 3.x release.
Referring to _the same left-hand parameter type_ is confusing imo.
Perhaps change it to:
_At present, `Collective extensions` have a limitation: the individual extension methods cannot be generic (i.e. they cannot have additional type parameters)._